test/date_time_parse_test.exs

```diff
@@ -3,5 +3,14 @@ defmodule DateTimeParseTest do
   alias Calendar.DateTime.Parse
   import Parse
   doctest Parse
-end
 
+  test "microsecond precision is capped at 6" do
+    assert(unix!("1000000000.123456789").microsecond == {123457, 6})
```

Comment:
Thanks, in general it looks good. For consistency we should not round up, could you do it so that this will return {123456, 6} instead? An issue with rounding is that e.g. 1000000000.99999999 becomes 1000000001.0. Rounding could change the year in some cases.